packages/content-management/table_list/src/reducer.tsx

```diff
@@ -0,0 +1,153 @@
+        field: 'updatedAt' as keyof T,
+        direction: 'desc' as const,
+      },
+      tableColumns: [
+        {
+          field: 'updatedAt',
+          name: i18n.translate('contentManagement.tableList.lastUpdatedColumnTitle', {
+            defaultMessage: 'Last updated',
+          }),
+          render: (field: string, record: { updatedAt?: string }) => (
+            <UpdatedAtField dateTime={record.updatedAt} DateFormatterComp={DateFormatterComp} />
+          ),
+          sortable: true,
+          width: '150px',
+        },
+      ],
+    };
+  }
+
+  return {};
+}
+
+export function getReducer<T extends UserContentCommonSchema>({ DateFormatterComp }: Dependencies) {
+  return (state: State<T>, action: Action<T>): State<T> => {
+    switch (action.type) {
+      case 'onFetchItems': {
+        return {
+          ...state,
+          isFetchingItems: true,
+        };
+      }
+      case 'onFetchItemsSuccess': {
+        const items = action.data.response.hits;
+        // We only get the state on the initial fetch of items
+        // After that we don't want to reset the columns or change the sort after fetching
+        const { tableColumns, tableSort } = state.hasInitialFetchReturned
+          ? { tableColumns: undefined, tableSort: undefined }
+          : onInitialItemsFetch(items, { DateFormatterComp });
+
+        return {
+          ...state,
+          hasInitialFetchReturned: true,
+          isFetchingItems: false,
+          items: !state.searchQuery ? sortBy<T>(items, 'title') : items,
+          totalItems: action.data.response.total,
+          tableColumns: tableColumns
+            ? [...state.tableColumns, ...tableColumns]
+            : state.tableColumns,
+          tableSort: tableSort ?? state.tableSort,
+          pagination: {
+            ...state.pagination,
+            totalItemCount: items.length,
+          },
+        };
+      }
+      case 'onFetchItemsError': {
+        return {
+          ...state,
+          hasInitialFetchReturned: true,
+          isFetchingItems: false,
+          items: [],
+          totalItems: 0,
+          fetchError: action.data,
+        };
+      }
+      case 'onSearchQueryChange': {
+        return {
+          ...state,
+          searchQuery: action.data,
+          isFetchingItems: true,
+        };
+      }
+      case 'onTableChange': {
+        const tableSort = action.data.sort ?? state.tableSort;
+        return {
+          ...state,
+          pagination: {
+            ...state.pagination,
+            pageIndex: action.data.page.index,
+            pageSize: action.data.page.size,
+          },
+          tableSort,
+        };
+      }
+      case 'showConfirmDeleteItemsModal': {
+        return {
+          ...state,
+          showDeleteModal: true,
+        };
+      }
+      case 'onDeleteItems': {
+        return {
+          ...state,
+          isDeletingItems: true,
+        };
+      }
+      case 'onCancelDeleteItems': {
```

Comment:
IMHO, this reducer shouldn't have so many responsibilities. I think a better architecture would be where one reducer would be handling actions for data fetching, the other items actions and the third table actions.

Reply:
Thanks for the input. I'll consider it (and refactor) when the state will get more complex. It might be a good approach.